internal/pkg/config/v1/config.go

```diff
@@ -647,8 +651,12 @@ func (c *Config) GetFilename() string {
 
 func (c *Config) CheckIsOnPremLogin() error {
 	ctx := c.Context()
-	if ctx != nil && ctx.PlatformName != "" && !c.isCloud() {
-		return nil
+	if ctx != nil && ctx.PlatformName != "" {
+		if !c.isCloud() {
+			return nil
+		} else {
+			return CloudLoginRequireOnPremLoginErr
```

Comment:
We should still print this if the user isn't logged in to **either** Cloud or CP.

Reply:
it'll be handled by `CheckIsNonAPIKeyCloudLoginOrOnPremLogin()`. This way i'm keeping the original logic and only adding a smaller branch